src/main/java/seedu/address/logic/LogicManager.java

```diff
@@ -69,16 +69,21 @@ private void assignStalledComponents(Command command, String commandText) {
     @Override
     public CommandResult execute(String commandText) throws CommandException, ParseException {
         logger.info("----------------[USER COMMAND][" + commandText + "]");
-        boolean isCancel = stalledCommand != null && !commandText.equalsIgnoreCase("yes");
+        Command command;
+        CommandResult commandResult;
 
-        if (isCancel) {
-            stalledCommand = null;
-            return new CommandResult(MESSAGE_CONFIRMATION_CANCEL
+        try {
+            command = gradPadParser.parseCommand(commandText);
+        } catch (ParseException e) {
+            if (stalledCommand != null) {
+                stalledCommand = null;
+                return new CommandResult(MESSAGE_CONFIRMATION_CANCEL
                     + String.format("\"%s\"", stalledCommandText));
+            } else {
+                throw e;
+            }
         }
 
-        CommandResult commandResult;
-        Command command = gradPadParser.parseCommand(commandText);
         boolean isConfirmation = command instanceof YesCommand && stalledCommand != null;
 
         if (command.requiresStall()) {
```

Comment:
We need to tweak the logic down here too cos now stalling doesn't work correctly :/
If you type in a valid command while the app is asking for confirmation, it will execute that command instead of aborting the current stalled command since we're only aborting it when we get a `ParseException` now.

E.g. `delete cs2101` --> (app asks for confirmation) --> `add cs1101s` --> (app adds cs1101s) --> `yes` --> (app still deletes cs2101

Reply:
ah man..... ok i'll tweak!